File: nvflare/edge/ios/NVFlareMobile/NVFlareMobile/Sources/Models/Connection/ResultResponse.swift
//
//  ResultResponse.swift
//  NVFlareMobile
//
//

import Foundation

struct ResultResponse: Decodable {
    let status: String
    let taskId: String?
    let taskName: String?
    let jobId: String?
    let message: String?
    let details: [String: String]?
    
    enum CodingKeys: String, CodingKey {
        case status
        case taskId = "task_id"
        case taskName = "task_name"
        case jobId = "job_id"
        case message
        case details
    }
}

File: examples/advanced/edge/mobile/android/sdk/models/ResultResponse.kt
package com.nvidia.nvflare.sdk.models

import com.google.gson.annotations.SerializedName
import com.google.gson.JsonObject

data class ResultResponse(
    @SerializedName("status")
    val status: String,
    
    @SerializedName("task_id")
    val taskId: String?,
    
    @SerializedName("task_name")
    val taskName: String?,
    
    @SerializedName("job_id")
    val jobId: String?,
    
    @SerializedName("message")
    val message: String?,
    
    @SerializedName("details")
    val details: Map<String, String>?
) {
    companion object {
        fun fromJSON(json: JsonObject): ResultResponse {
            return ResultResponse(
                status = json.get("status").asString,
                taskId = json.get("task_id")?.asString,
                taskName = json.get("task_name")?.asString,
                jobId = json.get("job_id")?.asString,
                message = json.get("message")?.asString,
                details = json.getAsJsonObject("details")?.let { details ->
                    val map = mutableMapOf<String, String>()
                    details.entrySet().forEach { (key, value) ->
                        map[key] = value.asString
                    }
                    map
                }
            )
        }
    }
}
